fabric-textures-v0/src/main/java/net/fabricmc/fabric/mixin/client/texture/MixinSpriteAtlasTexture.java

```diff
@@ -51,33 +53,38 @@
 	@Shadow
 	public abstract Sprite getSprite(Identifier id);
 
+	private final Set<Identifier> fabric_localIds = new HashSet<>();
+
+	// EVENT/HOOKS LOGIC
+
+	@Override
+	public void onRegisteredAs(Identifier id) {
+		fabric_localIds.add(id);
+	}
+
+	// INJECTION LOGIC
+
 	private Map<Identifier, Sprite> fabric_injectedSprites;
 
-	/**
-	 * The purpose of this patch is to allow injecting sprites at the stage of Sprite instantiation, such as
-	 * Sprites with CustomSpriteLoaders.
-	 * <p>
-	 * FabricSprite is a red herring. It's only use to go around Sprite's constructors being protected.
-	 * <p>
-	 * method_18160 is a lambda used in runAsync.
-	 */
-	@SuppressWarnings("JavaDoc")
-
-	@Redirect(method = "method_18160", at = @At(value = "NEW", target = "net/minecraft/client/texture/Sprite"))
-	public Sprite newSprite(Identifier id, PngFile pngFile, AnimationResourceMetadata animationMetadata) {
-		if (fabric_injectedSprites.containsKey(id)) {
-			return fabric_injectedSprites.get(id);
-		} else {
-			return new FabricSprite(id, pngFile, animationMetadata);
+	// Loads in custom sprite object injections.
+	@Inject(at = @At("RETURN"), method = "loadSprites")
+	private void afterLoadSprites(ResourceManager resourceManager_1, Set<Identifier> set_1, CallbackInfoReturnable<Collection<Sprite>> info) {
+		if (fabric_injectedSprites != null) {
+			info.getReturnValue().addAll(fabric_injectedSprites.values());
+			fabric_injectedSprites = null;
 		}
 	}
 
+	// Handles DependentSprite + custom sprite object injections.
 	@ModifyVariable(at = @At(value = "INVOKE", target = "Lnet/minecraft/client/texture/SpriteAtlasTexture;loadSprites(Lnet/minecraft/resource/ResourceManager;Ljava/util/Set;)Ljava/util/Collection;"), method = "stitch")
-	public Set<Identifier> setHook(Set<Identifier> set) {
-		fabric_injectedSprites = new HashMap<>();
+	public Set<Identifier> beforeSpriteLoad(Set<Identifier> set) {
+		fabric_injectedSprites = new LinkedHashMap<>();
```

Comment:
Just curious - why a LinkedHashMap here?  Does insertion order matter?

Reply:
Frankly speaking, not really. Mods should use DependentSprite. I guess I'll change it back to a HashMap.